### DISCUSSION TITLE
主观对比一下各种协议

### DISCUSSION THREAD

**Discussion by CodingMoeButa** [v2fly/v2ray-core#3016] [Show and tell]
- Shadowsocks直捣黄龙，直接解决问题，不需要配置TLS证书，很方便，虽然以前出过漏洞，但现在也没有谣传的那么容易被封。
- VMess设计比较独特，算法的扩展性比较差，现在感觉用的人少了，而且不如Shadowsocks来得直接。
- VLESS已经被V2Fly抛弃了，跟Trojan本质是一样的。
- Trojan是个模仿HTTPS的实现，需要配置证书，混淆性比较好。
- 但是说起HTTPS，不得不提Hysteria，Hysteria实现的QUIC不就是HTTP 3吗，这不比Trojan更前瞻。
- V2Ray实现的Websocket传输层，和CDN一起用比较合适。
- V2Ray实现的QUIC传输层，不知道怎么，速度很慢。
- V2Ray实现的mKCP传输层，我自己用着比QUIC传输层快得不是一点半点，作为基于UDP的协议还是很不错的，不过比不过Hysteria，而且除了v2ray-core和xray-core就没别的内核支持它了。
- VLESS+h2c+Caddy是个小众用法，不过挺稳的，就是结构比较麻烦。
- TUIC开发者已经删库了，而且这个协议本身也只是普通的QUIC实现，不如Hysteria。
- Naive不就是为了隐藏TLS指纹吗，感觉不是很必要，用Trojan大概率不会有事。
- ShadowTLS适用于SNI阻断的地区，不过感觉可以用Shadowsocks或VMess吧，这两个又不必要TLS，感觉也没什么必要用它。

合并同类项一下，感觉Shadowsocks和Hysteria比较好，一个是TCP一个是UDP，互补了属于是。v2ray-core合并Hysteria 2的进度有点慢了，那个PR好像有两年了，太慢了。

**Comment by RPRX** [comment]
~~一言难尽，别的社区的 discussion 之类的从来不会让我失望~~
